src/main/java/com/fasterxml/jackson/databind/util/clhm/ConcurrentLinkedHashMap.java

```diff
@@ -0,0 +1,1597 @@
+     * page replacement policy. This is represented by a negative weight.
+     *
+     * Dead: The entry is not in the hash-table and is not in the page replacement
+     * policy. This is represented by a weight of zero.
+     *
+     * The Least Recently Used page replacement algorithm was chosen due to its
+     * simplicity, high hit rate, and ability to be implemented with O(1) time
+     * complexity.
+     */
+
+    /** The number of CPUs */
+    static final int NCPU = Runtime.getRuntime().availableProcessors();
+
+    /** The maximum weighted capacity of the map. */
+    static final long MAXIMUM_CAPACITY = Long.MAX_VALUE - Integer.MAX_VALUE;
+
+    /** The number of read buffers to use. */
+    static final int NUMBER_OF_READ_BUFFERS = ceilingNextPowerOfTwo(NCPU);
+
+    /** Mask value for indexing into the read buffers. */
+    static final int READ_BUFFERS_MASK = NUMBER_OF_READ_BUFFERS - 1;
+
+    /** The number of pending read operations before attempting to drain. */
+    static final int READ_BUFFER_THRESHOLD = 32;
+
+    /** The maximum number of read operations to perform per amortized drain. */
+    static final int READ_BUFFER_DRAIN_THRESHOLD = 2 * READ_BUFFER_THRESHOLD;
+
+    /** The maximum number of pending reads per buffer. */
+    static final int READ_BUFFER_SIZE = 2 * READ_BUFFER_DRAIN_THRESHOLD;
+
+    /** Mask value for indexing into the read buffer. */
+    static final int READ_BUFFER_INDEX_MASK = READ_BUFFER_SIZE - 1;
+
+    /** The maximum number of write operations to perform per amortized drain. */
+    static final int WRITE_BUFFER_DRAIN_THRESHOLD = 16;
+
+    /** A queue that discards all entries. */
+    static final Queue<?> DISCARDING_QUEUE = new DiscardingQueue();
+
+    static int ceilingNextPowerOfTwo(int x) {
+        // From Hacker's Delight, Chapter 3, Harry S. Warren Jr.
+        return 1 << (Integer.SIZE - Integer.numberOfLeadingZeros(x - 1));
+    }
+
+    // The backing data store holding the key-value associations
+    final ConcurrentMap<K, Node<K, V>> data;
+    final int concurrencyLevel;
+
+    // These fields provide support to bound the map by a maximum capacity
+    //@GuardedBy("evictionLock")
+    final long[] readBufferReadCount;
+    //@GuardedBy("evictionLock")
+    final LinkedDeque<Node<K, V>> evictionDeque;
+
+    //@GuardedBy("evictionLock") // must write under lock
+    final AtomicLong weightedSize;
+    //@GuardedBy("evictionLock") // must write under lock
+    final AtomicLong capacity;
+
+    final Lock evictionLock;
+    final Queue<Runnable> writeBuffer;
+    final AtomicLong[] readBufferWriteCount;
+    final AtomicLong[] readBufferDrainAtWriteCount;
+    final AtomicReference<Node<K, V>>[][] readBuffers;
+
+    final AtomicReference<DrainStatus> drainStatus;
+    final EntryWeigher<? super K, ? super V> weigher;
+
+    // These fields provide support for notifying a listener.
+    final Queue<Node<K, V>> pendingNotifications;
+    final EvictionListener<K, V> listener;
```

Comment:
We can get rid of notifications, listeners, too?

Reply:
the weigher could be removed if desired, since it is just evaluating to `1`, so could be striped out without much pain (I think)